src/measure_theory/function/conditional_expectation.lean

```diff
@@ -2092,6 +2092,23 @@ begin
       (signed_measure.measurable_rn_deriv _ _).strongly_measurable },
 end
 
+lemma condexp_mono {m : measurable_space α} [is_finite_measure μ] {f g : α → ℝ}
+  (hf : integrable f μ) (hg : integrable g μ) (hfg : f ≤ᵐ[μ] g) :
+  μ[f | m] ≤ᵐ[μ] μ[g | m] :=
```

Comment:
The "right" way to do this is through the construction of the conditional expectation using `set_to_fun`: we already had a mono lemma on `set_to_fun` and can use it, provided we prove that the linear map given to `set_to_fun` has a nonnegativity property (which is done very similarly to what you did here). I did that work in #15024. The final statement is this:
```lean
lemma condexp_mono {E} [normed_lattice_add_comm_group E] [complete_space E] [normed_space ℝ E]
  [ordered_smul ℝ E] {f g : α → E} (hf : integrable f μ) (hg : integrable g μ) (hfg : f ≤ᵐ[μ] g) :
  μ[f | m] ≤ᵐ[μ] μ[g | m] :=
```